src/index.ts

```diff
@@ -1,8 +1,136 @@
+  try {
+    container.loggerService.info('Loading and running Bot...');
+
+    try {
+      await listener.container.storageService.connectToDB();
+    } catch (e) {
+      container.loggerService.error(`Could not connect to db: ${e}`);
+    }
+
+    // register plugins
+    container.pluginService.reset();
+
+    // load classic plugins
+    const pluginFolder = path.join(__dirname, '/app/plugins');
+    const pluginFiles = await readdir(pluginFolder);
+    await Promise.allSettled(
+      pluginFiles.map(async (file) => {
+        // Make sure file is proper.
+        if (!file.endsWith('.ts') && !file.endsWith('.js')) {
+          return;
+        }
+
+        // Import the class from the plugin file.
+        const pluginInstance = await import(`./app/plugins/${file}`);
+
+        // Try to see if it's in the proper form.
+        try {
+          // Check constructor.
+          const plugin = new pluginInstance.default(container);
+
+          // Check instance.
+          if (!(plugin instanceof Plugin)) {
+            container.loggerService.error(
+              `${file} has a default export, but it is not of type Plugin`
+            );
+            return;
+          }
+
+          // Register plugin.
+          container.pluginService.register(plugin);
+        } catch (err) {
+          container.loggerService.warn(`${file} doesn't have a default export of type Plugin!`);
+        }
+      })
+    );
+
+    // load slash plugins
+    const slashPluginFolder = path.join(__dirname, '/app/slash_plugins');
+    const slashPluginFiles = await readdir(slashPluginFolder);
+    await Promise.allSettled(
+      slashPluginFiles.map(async (file) => {
+        if (!file.endsWith('.ts') && !file.endsWith('.js')) {
+          return;
+        }
+
+        const plugin = await import(`./app/slash_plugins/${file}`).then((m) => m.default);
+
+        const result = SlashCommand.safeParse(plugin);
+        // FIXME we could validate more here: for example, checking the parameters
+        // property, but bot should be fine for now
+        if (result.success) {
+          slashCommands.set(plugin.commandName, result.data as ISlashCommand);
+        } else {
+          container.loggerService.warn(
+            `${file} does not \`export default\` a plugin with type ISlashCommand!`
+          );
+          container.loggerService.warn(result.error);
+        }
+      })
+    );
+
+    const slashCommandUploads = Array.from(slashCommands.entries()).map(([key, command]) => {
+      return {
+        name: key,
+        description: command.description.substring(0, 99),
+        options: command.options,
+      };
+    });
+    // Register commands for all guilds.
+    await Promise.all(
+      container.clientService.guilds.cache.map((guild) => guild.commands.set(slashCommandUploads))
+    );
+
+    // register jobs
+    container.jobService.reset();
+
+    const jobs = container.jobService.jobs;
+    for (const job of jobs) {
```

Comment:
can you make this loop consistent w the stores loop? one is for-of the other is forEach

Reply:
Done